src/main/java/com/cloudbees/hudson/plugins/folder/AbstractFolder.java

```diff
@@ -0,0 +1,746 @@
+
+    /**
+     * Currently active Views tab bar.
+     */
+    private volatile ViewsTabBar viewsTabBar;
+
+    /**
+     * Name of the primary view.
+     */
+    private volatile String primaryView;
+
+    private transient /*almost final*/ ViewGroupMixIn viewGroupMixIn;
+
+    private DescribableList<FolderHealthMetric,FolderHealthMetricDescriptor> healthMetrics;
+
+    private FolderIcon icon;
+
+    /** Subclasses should also call {@link #init}. */
+    protected AbstractFolder(ItemGroup parent, String name) {
+        super(parent, name);
+    }
+
+    protected void init() {
+        if (properties == null) {
+            properties = new DescribableList<AbstractFolderProperty<?>,AbstractFolderPropertyDescriptor>(this);
+        }
+        for (AbstractFolderProperty p : properties) {
+            p.setOwner(this);
+        }
+        if (icon == null) {
+            icon = new StockFolderIcon();
+        }
+        if (views == null) {
+            views = new CopyOnWriteArrayList<View>();
+        }
+        if (views.isEmpty()) {
+            try {
+                initViews(views);
+            } catch (IOException e) {
+                LOGGER.log(Level.WARNING, "Failed to set up the initial view", e);
+            }
+        }
+        if (viewsTabBar == null) {
+            viewsTabBar = new DefaultViewsTabBar();
+        }
+        if (primaryView == null) {
+            primaryView = views.get(0).getViewName();
+        }
+        viewGroupMixIn = new ViewGroupMixIn(this) {
+            @Override
+            protected List<View> views() {
+                return views;
+            }
+            @Override
+            protected String primaryView() {
+                return primaryView;
+            }
+            @Override
+            protected void primaryView(String name) {
+                primaryView = name;
+            }
+        };
+        if (healthMetrics == null) {
+            List<FolderHealthMetric> metrics = new ArrayList<FolderHealthMetric>();
+            for (FolderHealthMetricDescriptor d : FolderHealthMetricDescriptor.all()) {
+                FolderHealthMetric metric = d.createDefault();
+                if (metric != null) {
+                    metrics.add(metric);
+                }
+            }
+            healthMetrics = new DescribableList<FolderHealthMetric, FolderHealthMetricDescriptor>(this, metrics);
+        }
+    }
+
+    protected void initViews(List<View> views) throws IOException {
+        AllView v = new AllView("All", this);
+        views.add(v);
+        v.save();
+    }
+
+    @Override
+    public void onLoad(ItemGroup<? extends Item> parent, String name) throws IOException {
+        super.onLoad(parent, name);
+        init();
+        final Thread t = Thread.currentThread();
+        String n = t.getName();
+        try {
+            if (items ==null) {
+                // when Jenkins is getting reloaed, we want children being loaded to be able to find
```

Comment:
typo in `reloaed` also is the comment still valid? Does `loadChildren` respect the existing entries?